addons/mrp/views/mrp_production_views.xml

```diff
@@ -101,6 +101,28 @@
             <field name="code">action = records.action_split()</field>
         </record>
 
+        <record id="action_production_order_lock_unlock" model="ir.actions.server">
+            <field name="name">Lock/Unlock</field>
+            <field name="model_id" ref="mrp.model_mrp_production"/>
+            <field name="binding_model_id" ref="mrp.model_mrp_production"/>
+            <field name="binding_view_types">list,form</field>
+            <field name="state">code</field>
+            <field name="code">
+            if records:
+                records.action_toggle_is_locked()</field>
+        </record>
+
+        <record id="action_production_order_scrap" model="ir.actions.server">
+            <field name="name">Scrap</field>
+            <field name="model_id" ref="mrp.model_mrp_production"/>
+            <field name="binding_model_id" ref="mrp.model_mrp_production"/>
+            <field name="binding_view_types">list,form</field>
+            <field name="state">code</field>
+            <field name="code">
+            if records:
+                action = records.button_scrap()</field>
```

Comment:
`button_scrap` still begins with `self.ensure_one()` so won't work in list view

Reply:
Oh my bad, `button_scrap` is not needed in the list view too